src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -328,37 +321,22 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 	public onNextChannelResponse = (messages: EventMessage[]) => {
 		this.lastNextChannelResponseTimestamp = null;
 
-		// eslint-disable-next-line no-param-reassign
-		messages = messages.filter(
-			msg => msg.messageId !== this.messagesStore.selectedMessageId?.valueOf(),
-		);
-
-		const prevMessages =
-			this.messages.length > 0
-				? messages
-						.filter(
-							message =>
-								timestampToNumber(message.timestamp) <
-								timestampToNumber(this.messages[0].timestamp),
-						)
-						.sort((a, b) => timestampToNumber(b.timestamp) - timestampToNumber(a.timestamp))
-				: [];
-
 		const nextMessages = messages
-			.slice(0, messages.length - prevMessages.length)
+			.filter(message => !this.messages.map(msg => msg.messageId).includes(message.messageId))
```

Comment:
instead of creating an array of ids for every message, you could have used .find method

if there is a problem about duplicating messages even a better way to do this would be to have a computed value Map<messageId, true> from messages list, so this check would be more efficient

Reply:
So there is no duplicates problem as far as I see now, so I'll leave it with .find, if the bug will appear in future will use Map<messageId, true>